### PR TITLE
Tooling - 4027 - Add storybook dark mode

### LINKED ISSUE #4027: Add dark mode to storybook
We need to be able to see components in dark mode in storybook.

Tristan:  "I think we can use a [decorator](https://storybook.js.org/docs/react/writing-stories/decorators) and a [global ](https://storybook.js.org/docs/react/essentials/toolbars-and-globals)to add a light/dark switcher right to the Storybook toolbar."

Acceptance Criteria:
- [ ] Can interactively switch between light and dark mode for stories
- [ ] Can define stories in dark mode
- [ ] Can snapshot for Chromatic in dark mode

Out of Scope:
- [ ] Responsibility for testing homepage in dark mode is #4028

### PR DESCRIPTION
**Pull request by esizer**
Resolves #4027 

> **Note**
> This was in the safebox but I already had most of it stashed, just needed to wire things up.

## Summary

This adds a global control to storybook for our `ThemeProvider` for dark mode 🌔 .

## Screenshot

Screen Shot 2022-10-06 at 1 02 38 PM

## Testing

> **Note**
> The best stories to test this are "ThemeSwitcher" and "Home Page" (GC Digital, not IAP).

1. Startup storybook `npm run storybook`
2. Confirm the global control exists in the header
3. Confirm it actually changes the theme mode

### CODE REVIEW

**Comment by petertgiles** [comment]
This works interactively!  Cool!
Is it possible to specify a specific theme for a story to have it come like that by default?  And how about a specific theme for snapshots?  It would be cool to have a light and a dark snapshot for the new homepage.

**Comment by esizer** [comment]
> This works interactively! Cool! Is it possible to specify a specific theme for a story to have it come like that by default? And how about a specific theme for snapshots? It would be cool to have a light and a dark snapshot for the new homepage.
> 
> I see there's a override prop. Is that it?

@petertgiles Got it working in chromatic and only chromatic. You should now 

> And how about a specific theme for snapshots? It would be cool to have a light and a dark snapshot for the new homepage.

It's working! 

Its working

**Comment by esizer** [comment]
> Is it possible to specify a specific theme for a story to have it come like that by default?

Sorry, forgot to answer this question. Yes, it is possible to set the default by using paramters.

```tsx
export const Default = Template.bind({});
Default.parameters = {
  theme: "dark",
};
```

File: frontend/common/.storybook/decorators/ThemeDecorator.tsx
Comment (petertgiles) [inline frontend/common/.storybook/decorators/ThemeDecorator.tsx:34]: Unused interface?

**Review by petertgiles — CHANGES_REQUESTED** [review]
Oh man, this is so cool.  The Chromatic diffs look great.  But now the homepage won't swap for me in Storybook.  :cry: 

https://user-images.githubusercontent.com/8978655/194629712-3718223a-337c-427e-a055-e891bd9bb39f.mp4

File: frontend/common/.storybook/decorators/ThemeDecorator.tsx
Comment (esizer) [reply frontend/common/.storybook/decorators/ThemeDecorator.tsx]: Removed with [remove unused interface](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4229/commits/aa9b1c29e96ab6d95fdb41548888433c6ef77d8f)

And, you can probably guess what was causing the home page forced to dark with the name of the fix: [revert theme decorator debugging](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4229/commits/0a03049131f69292bc4396974b0d08e5e0209b48)

**Review by petertgiles — APPROVED** [review]
Love it!  :artist: :dark_sunglasses: :bat:

**Comment by esizer** [comment]
> Love it! 🧑‍🎨 🕶️ 🦇

Just waiting on chromatic before a merge incase that debug code was somehow making it work 😆

**Comment by JoshBeveridge** [comment]
I know this is closed but YAAAAAAAAAAAAAAA 🥳

**Pull request merged**
